build/helper.py

```diff
@@ -136,23 +143,38 @@ def add_all_metadata(functions):
             _add_buffer_info(p)
     return functions
 
-# Normalize string type between python2 & python3
+# Python 2/3 compatibility
+
 def normalize_string_type(d):
+    '''Normalize string type between python2 & python3'''
     if sys.version_info.major < 3:
         if type(d) is dict:
             for k in d:
                 d[k] = normalize_string_type(d[k])
         elif type(d) is str:
             d = d.decode('utf-8')
-
     return d
+
 # Functions that return snippets that can be placed directly in the templates.
 
-def get_method_parameters_snippet(parameters):
-    '''Returns a string suitable for the parameter list of a method given a list of parameter objects'''
+def get_method_parameters_snippet(parameters, skip_session_handle, skip_ivi_dance_size_parameter, skip_output_parameters):
+    '''
+    Returns a string suitable for the parameter list of a method given a list of parameter objects.
+    You can optionally skip session handle parameter, the parameter used for an output
+    buffer size (i.e. you don't want it in a Session method), and output parameters.
+    '''
     snippets = ['self']
+    ivi_dance_size_parameter = find_size_parameter(extract_ivi_dance_parameter(parameters), parameters)
```

Comment:
find_size_parameter() uses ['size']['value'] as the name to search for in find_parameter() which will assert if nothing is found. 

You call it here unconditionally, so the default value of 1 can be used, right? Or am I missing something?